developing_packages/editable_packages.rst

```diff
@@ -0,0 +1,224 @@
+        settings = "os", "compiler", "arch", "build_type"
+        def package_info(self):
+            if not self.in_local_cache:
+                if self.settings.compiler == "Visual Studio":
+                    # NOTE: Use the real layout used in your VS projects, this is just an example
+                    self.cpp_info.libdirs = ["%s_%s" % (self.settings.build_type, self.settings.arch)]
+
+That will define the libraries directories to ``path/to/local/dev/libcool/Release_x86_64``, for example.
+That is only an example, the real layout used by VS would be different.
+
+Layout files
++++++++++++++
+
+Layout files have the following syntax:
+
+   .. code-block:: ini
+
+       [includedirs]
+       src/core/include
+       src/cmp_a/include
+
+       [libdirs]
+       build/{settings.build_type}/{settings.arch}
+
+       [bindirs]
+       build/{settings.build_type}/{settings.arch}
+
+As you can see, you can use some **placeholders** inside these files that will be substituted with
+the values of the ``settings`` and the ``options`` of the package.
+
+This file can use the package reference to customize logic for a specific package:
+
+   .. code-block:: ini
+
+       [includedirs]
+       src/include
+
+       [cool/version@user/dev:includedirs]
+       src/core/include
+
+This layout will define the ``src/core/include`` include directory for the ``cool`` package, and
+``src/include`` for other packages in editable mode.
+
+In every case the directories that will be affected by the editable mode will be ``includedirs``,
+``libdirs``, ``bindirs``, ``resdirs``, ``srcdirs`` and ``builddirs``, all of them declared in the
+:ref:`cpp_info_attributes_reference` dictionary; the rest of values in that dictionary won't
+be modified. So ``cflags``, ``defines``, library names in ``libs`` defined in ``package_info()``
+will still be used.
+
+By default all folders paths are relative to the directory where the *conanfile.py*
+of the editable package is (which is the path used to create the link), though they also allow absolute
+paths.
+
+Specifying layout files
++++++++++++++++++++++++
+
+Layout files are specified in the :command:`conan link` command, as an extra argument:
+
+.. code-block:: bash
+
+    $ conan link . cool/version@user/dev --layout=win_layout
+
+That ``win_layout`` file will be first looked for relative to the current directory (the
+path can be absolute too). If it is found, that will be used. It is possible to add those
+layouts in the source repositories, so they are always easy to find after a clone.
+
+If the specified layout is not found relative to the current directory, it will be looked
+for in the conan cache, in the ``.conan/layouts`` folder. This is very convenient to have
+a single definition of layouts that can be shared with the team and installed with
+``conan config install``.
+
+If no argument is specified, the :command:`conan link` command will try to use a `.conan/layouts/default`
+layout from the local cache.
```

Comment:
I would add a comment about when the `Recipe defined layout` will be used if there is some kind of default to `.conan/layouts/default` file